app/services/invoices/issuing_date_service.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module Invoices
+  class IssuingDateService < BaseService
+    attr_reader :customer, :recurring
+
+    def initialize(customer:, recurring: false)
+      @customer = customer
+      @recurring = recurring
+      super
+    end
+
+    def base_date(timestamp)
+      date = if recurring && current_period_end?
+        timestamp - 1.day
+      else
+        timestamp
+      end
+
+      date.in_time_zone(customer.applicable_timezone).to_date
+    end
+
+    def grace_period
+      period = customer.applicable_invoice_grace_period
+
+      return period unless recurring
+      return 0 if keep_anchor?
+
+      current_period_end? ? period + 1 : period
```

Comment:
I believe this logic might be incorrect. `current_period_end` should have no impact here.

You can actually see the impact in the `spec/services/invoices/create_generating_service_spec.rb` where `subscription_invoice_issuing_date_anchor` has no impact on the issuing date whether `subscription_invoice_issuing_date_anchor` is `current_period_end` or `next_period_start`.